open_dread_rando/dread_patcher.py

```diff
@@ -77,6 +78,10 @@ def chunks(array, n):
             box_text = "|".join(box)
             patch_text(editor, f"RANDO_STARTING_TEXT_{textboxes}", box_text)
 
+    layout_uuid = None
+    if "layout_uuid" in configuration:
+        layout_uuid = str(uuid.UUID(configuration["layout_uuid"]).bytes)[1:]
```

Comment:
Tbh I don't get the purpose of that. This will create a string with escaped bytes. 
Why don't we just put the hex string in it as it is?

Reply:
if nothing else, this is ugly and arcane and I'd prefer another solution be used for that reason only

Reply:
I agree with Thanatos that it makes more sense to just add a hex string, unless there's some lua limitation i'm unaware of. in which case you could simplify that to `lua_util.wrap_string(uuid.UUID(configuration["layout_uuid"]).hex)`.

Reply:
Doesn't even the schema with the string pattern already make sure that `configuration["layout_uuid"]` is a valid uuid? Then it's really only `configuration["layout_uuid"]`

Reply:
> Doesn't even the schema with the string pattern already make sure that `configuration["layout_uuid"]` is a valid uuid? Then it's really only `configuration["layout_uuid"]`

That would probably work as well, I just left the conversion from uuid to hex in case removing the hyphens was important for lua run on the console. I assume that it's just used for sending data to the server, in which case `uuid.UUID()` would get the same uuid from either value.

Reply:
It's a case of minimising how much data is sent. An uuid is 16 bytes, but using the string representation takes 36 ASCII characters which would take 36 bytes to be sent.

Lua strings are just size + data, so they're able of holding raw bytes. My goal is to put the direct bytes of the uuid here so when sent over the network it's smaller.

Reply:
I get the idea but the current bootstrap code to connect to dread in a multiworld context is like 3 messages with around 8 kB.
I personally would prefer a more readable string instead of saving 20 bytes.